ncp-web/ncp-launcher.php

```diff
@@ -80,13 +80,33 @@
   }
 
   // launch
-  echo '{ "token": "' . getCSRFToken() . '",';     // Get new token
-  echo ' "ref": "' . $ncp_app          . '",';
-  echo ' "output": "" , ';
-  echo ' "ret": ';
 
-  exec( 'bash -c "sudo /home/www/ncp-launcher.sh ' . $ncp_app . '"' , $output , $ret );
-  echo '"' . $ret . '" }';
+  $ret = null;
+  $output = null;
+  if( file_exists("/usr/local/etc/ncp.lock") )
+  {
+    $running = trim(file_get_contents("/usr/local/etc/ncp.lock"));
+    $output = "";
+    $output .= "An app ($running) is already running...".PHP_EOL;
+    if ( file_exists("/var/log/ncp/tmux.$running.log") )
+    {
+      $output .= "Attaching to its output:".PHP_EOL;
+      $output .= file_get_contents("/var/log/ncp/tmux.$running.log");
+    }
+    $ret=-1;
+  }
+  else
+  {
+    exec( 'sudo /home/www/ncp-launcher.sh ' . $ncp_app , $output , $ret );
+    $output = "";
```

Comment:
```suggestion
```

Reply:
Fixed in cd8451d318815945bae42ddda0a16b0a1ff1468a